src/qibocal/auto/operation.py

```diff
@@ -189,25 +188,14 @@ def update(self) -> dict[str, ParameterValue]:
 
     def save(self, path):
         """Store results to json."""
-        result_dict = {}
-        convert = Key()
-        for result_name, result in asdict(self).items():
-            result_dict[result_name] = {}
-            if isinstance(result, dict):
-                for key, elem in result.items():
-                    result_dict[result_name][convert.dump(key)] = elem
-
-            else:
-                result_dict[result_name] = result
-        (path / RESULTSFILE).write_text(json.dumps(result_dict, indent=4))
+        (path / RESULTSFILE).write_text(json.dumps(serialize(asdict(self)), indent=4))
```

Comment:
```suggestion
        (path / RESULTSFILE).write_text(json.dumps(serialize(asdict(self))))
```
These files are not for manual inspection (mainly, at least). And it's always possible to easily reformat a JSON, even with standard CLI tools like `jq`.

Consider that each 4 spaces are 4 bytes (in UTF-8), so it would increase the size of data just for presentation purposes.
Let's reduce the required storage, and format on-demand.